src/metabase/models/card.clj

```diff
@@ -840,6 +855,64 @@
     ;; Some sort of odd query like an aggregation over cards. Just return it as-is.
     card))
 
+(defonce ^{:doc "Set of card IDs which have been read, which are in `:not-started` state.
+           These cards should be prioritized for analysis."}
+  cards-for-priority-analysis
+  (atom #{}))
+
+(def ^:dynamic *upstream-cards-without-idents*
+  "Set of card IDs which were read during analysis of another card and which did not have valid idents
+  already. In other words, this is the set of cards which are blocking a :blocked card."
+  nil)
+
+(defn- mark-card-used-for-priority-analysis
+  "If we selected `:metadata_analysis_state` and it's `:not-started` then add this card's ID to the set of cards
+  we want to prioritize for analysis."
+  [{state :metadata_analysis_state :as card}]
+  ;; Mark :not-started cards as eligible for :priority analysis.
+  (when (= state :not-started)
+    (swap! cards-for-priority-analysis conj (:id card)))
```

Comment:
This seems like a good spot to enforce a maximum size on the set.

Reply:
I made that change; now we won't store anything in `cards-for-priority-analysis` unless the job is running.

Reply:
It's less the size that I worry about, and more the performance and garbage of such a massive persistent structure. Even though we're populating it in batches, my concern is that some other background process sweeps across the cards more rapidly than this backfill, hitting that hook.

Feel free to resolve if you're not concerned, just nice to have a belt along with suspenders.